torch/ao/quantization/fx/_decomposed.py

```diff
@@ -319,7 +319,8 @@ def choose_qparams_tensor_meta(
         eps: float,
         dtype: torch.dtype
 ) -> Tuple[torch.Tensor, torch.Tensor]:
-    assert input.dtype == torch.float32, f"Expecting input to have dtype torch.float32, but got dtype: {input.dtype}"
+    assert input.dtype in (torch.float32, torch.float64), \
```

Comment:
Why we have `torch.float64` generated here?

Reply:
Maybe it is due to type promotion as we discussed offline.

Reply:
I think we may need some further study for this type promotion, since the behavior looks different as eager.

Reply:
Any case that causes the type promotion to `torch.float64`?

Reply:
Hi @jgong5 `choose_qparams` returns scale as a float64 tensor. The scale tensor is used by `dequantize_per_tensor`, whose output is then used by `aten.linear`.  In the UT, there are two linear layers. The linear output will be used by another `choose_qparams`.
In the implementation of `dequantize_per_tensor` here https://github.com/pytorch/pytorch/blob/be31e522ce33f8fc5d53dd0681e9e0f796dd7a90/torch/ao/quantization/fx/_decomposed.py#L198 the value of scale is got by `scale.item()`. The `item()` method will return a float32 value even if scale is a float64 tensor. So, in this case (no lowering to Inductor), no float64 is seen later on.

However, we registered `dequantize_per_tensor` for Inductor lowering  as https://github.com/pytorch/pytorch/blob/be31e522ce33f8fc5d53dd0681e9e0f796dd7a90/torch/_inductor/decomposition.py#L529 where the `item()` method is not used because it cannot be lowered. The output dtype is float64. So, in this case, the data type promotion will be carried on to subsequent ops, including the `choose_qparams` here.

There are two solutions. One is to allow float64 here. The other is to convert scale to float32 before this line https://github.com/pytorch/pytorch/blob/be31e522ce33f8fc5d53dd0681e9e0f796dd7a90/torch/_inductor/decomposition.py#L529
The two solutions are actually two different kinds of behavior. We are not sure which which behavior is preferred. Which do you think is better? Thanks.

Reply:
It seems converting `scale` to `float32` in the decomposition is more consistent?

Reply:
Ok. Thanks. I have fixed it in this way.